src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -52,104 +67,834 @@ static ColliderJntSphInit D_809BF3F8 = {
+            this->unk_196 = 4;
+        } else {
+            this->unk_196 = 0;
+        }
+    }
+    this->unk_19A = 0;
+    this->collider.base.acFlags &= ~1;
+    this->actionFunc = func_809BE3E4;
+}
+
+void func_809BD768(EnBigokuta* this) {
+    this->unk_194 = (Rand_ZeroOne() < 0.5f) ? -1 : 1;
+    this->unk_19A = 0;
+    this->actor.flags &= ~1;
+    this->cylinder[0].base.atFlags &= ~1;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_SINK);
+    this->actionFunc = func_809BE4A4;
+}
+
+void func_809BD7F0(EnBigokuta* this, GlobalContext* globalCtx) {
+    this->actor.world.rot.y = Actor_WorldYawTowardPoint(globalCtx->actorCtx.actorLists[2].head, &this->actor.home.pos);
+    this->actor.shape.rot.y = this->actor.world.rot.y + (this->unk_194 * 0x4000);
+    func_809BCE3C(this);
+    this->actionFunc = func_809BE518;
+}
+
+void func_809BD84C(EnBigokuta* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+
+    this->unk_196--;
+
+    if (this->unk_196 == 13 || this->unk_196 == -20) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_VOICE);
+    }
+    if (this->unk_196 == 1) {
+        func_800F5ACC(0x38);
+    }
+    if (this->actor.params == 1) {
+        func_809BD370(this);
+    }
+}
+
+void func_809BD8DC(EnBigokuta* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+
+    this->unk_196--;
+
+    if (this->unk_196 >= 0) {
+        Math_StepToF(&this->actor.world.pos.x, this->actor.home.pos.x + 263.0f, 263.0f / 21);
+
+        if (this->unk_196 < 14) {
+            this->actor.world.pos.y = sinf(this->unk_196 * (M_PI / 28)) * 200.0f + this->actor.home.pos.y;
+        } else {
+            this->actor.world.pos.y =
+                sinf((this->unk_196 - 7) * (M_PI / 14)) * 130.0f + (this->actor.home.pos.y + 70.0f);
+        }
+        if (this->unk_196 == 0) {
+            effectPos.x = this->actor.world.pos.x + 40.0f;
+            effectPos.y = this->actor.world.pos.y;
+            effectPos.z = this->actor.world.pos.z - 70.0f;
+            EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 1, 2000);
+            effectPos.x = this->actor.world.pos.x - 40.0f;
+            EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 1, 2000);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_LAND_WATER);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOLON_LAND_BIG);
+            func_800AA000(0.0f, 0xFF, 0x14, 0x96);
+        }
+    } else if (this->unk_196 < -1) {
+        this->actor.world.pos.y = this->actor.home.pos.y - (sinf((this->unk_196 + 1) * (M_PI / 10)) * 20.0f);
+        if (this->unk_196 == -10) {
+            func_809BD3AC(this);
+        }
+    }
+}
+
+void func_809BDAE8(EnBigokuta* this, GlobalContext* globalCtx) {
+    if (Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y + 0x4000, 0x400)) {
+        if (this->unk_196 != 0) {
+            this->unk_196--;
+        }
+        if (this->unk_196 == 0) {
```

Comment:
DECR?

Reply:
Not OK if i move DECR in the if statement